template_library/views.py

```diff
@@ -0,0 +1,31 @@
+from allauth.account.views import ConfirmEmailView
+from allauth.socialaccount.providers.facebook.views import FacebookOAuth2Adapter
+from allauth.socialaccount.providers.google.views import GoogleOAuth2Adapter
+from django.shortcuts import redirect
+from rest_auth.registration.views import SocialLoginView
+
+
+class FacebookLogin(SocialLoginView):
+    adapter_class = FacebookOAuth2Adapter
+
+
+class GoogleLogin(SocialLoginView):
+    adapter_class = GoogleOAuth2Adapter
+
+
+class CustomConfirmEmailView(ConfirmEmailView):
+    """
+    Override default register view to
+    redirect users to create organizations
+    after signup
+    """
+
+    def post(self, *args, **kwargs):
+        redirect_url = 'list_orgs'
```

Comment:
this doesn't look right, it looks like are we redirecting to a URL that is just `list_orgs`, can you expand on this?

Reply:
the url points to a list/create view. the GET request will return an empty list, will use this information to load the form in the frontend. this form will be POST to the list_orgs. to create the orgs. I think the name should have been `list_create_orgs` 

> where are the following endpoints?
> 
> * password forgotten
> * change password
> * login
> * logout

These API have been implemented in the `rest-auth`. so from [here](https://django-rest-auth.readthedocs.io/en/latest/api_endpoints.html), i just need to call them from the frontend with the respective Methods

Reply:
I still don't get it, if you do redirect('list_orgs'), won't the user be redirected to `http://list_orgs`?

Reply:
`list_orgs` is a view name found in `openwisp_users`  i am passing a view name to redirect and not a url path